pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -950,6 +952,22 @@ public CompletableFuture<ClientCnx> getConnection(final String topic) {
                 .thenCompose(pair -> getConnection(pair.getLeft(), pair.getRight()));
     }
 
+    public CompletableFuture<ClientCnx> getConnection(final String topic, final String url) {
+        TopicName topicName = TopicName.get(topic);
+        return getLookup(url).getBroker(topicName).thenCompose(pair -> getConnection(pair.getLeft(), pair.getRight()));
```

Comment:
Does this cause more unnecessary lookup execution?

Can you explain the specific process of your blue and green release?